rack_tamagotchi/handlers/h_pet.rb

```diff
@@ -0,0 +1,54 @@
+module Handlers
+  require_relative 'base'
+
+  class Pet < Base
+
+    def initialize(request)
+      @request = request
+    end
+
+    def response
+      if @request.path_info.match /\A#{'/api/pets/'}\d+/
```

Comment:
split to methods
```ruby
if @request.path_info.match /\A#{'/api/pets/'}\d+/
  respond_pet
elsif @request.get? && @request.path_info.start_with?(/\A#{'/api/pets'}\z/)
  respond_pets
elsif @request.post? && @request.path_info.start_with?(/\A#{'/api/pets'}\z/)
  create_pet
else
  respond_404
end
```

Reply:
and now, in addition, you can move conditions to methods
```ruby
if respond_pet?
  respond_pet
elsif respond_pets?
  respond_pets
elsif create_pet?
  create_pet
else
  respond_404
end
```

Reply:
And now, using some metaprogramming
```ruby
HANDLERS = %i[respond_pet respond_pets create_pet].freeze

def response
  responded = false
  HANDLERS.each do |handler|
    if send("#{handler}?")
      send(handler)
      responded = true
     break
    end
  end
  respond_404 unless responded
end
```

So, now, if you must add new endpoint, you just have to implement two methods: checker and handler